pyro/mesh/patch.py

```diff
@@ -884,5 +884,69 @@ def do_demo():
     mydata.pretty_print("a")
 
 
+# **c checking
+class PolarGrid(Grid2d):
+    """
+    the 2-d grid class.  The grid object will contain the coordinate
+    information (at various centerings).
+
+    A basic (1-d) representation of the layout is::
+
+       |     |      |     X     |     |      |     |     X     |      |     |
+       +--*--+- // -+--*--X--*--+--*--+- // -+--*--+--*--X--*--+- // -+--*--+
+          0          ng-1    ng   ng+1         ... ng+nx-1 ng+nx      2ng+nx-1
+
+                            ilo                      ihi
+
+       |<- ng guardcells->|<---- nx interior zones ----->|<- ng guardcells->|
+
+    The '*' marks the data locations.
+    """
+
+    # pylint: disable=too-many-instance-attributes
+
+    def area_x(self):
+        """
+        Return an array of the face areas. 
+        The shape of the returned array is (ni, nj).
+        """
+        r1 = self.xr
+        r0 = self.xl
+
+        # ** this is just \Delta r
+
+        area = r1 - r0
+        return area
+
+    def area_y(self):
+        """
```

Comment:
this should be the area through a y face, which is just dr

Reply:
but y face is dtheta